cpp/include/raft/matrix/select.cuh

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "detail/select_radix.cuh"
+#include "detail/select_warpsort.cuh"
+
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/nvtx.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <optional>
+
+namespace raft::matrix {
```

Comment:
I could go either way with this name. I think `select_h.cuh` might be a little easier to remember since the underlying function is `select_k`. On the other hand, do you foresee other functions being added to this file for different types of selection? I almost wonder if `argmax` and `argmin` might belong here as well? Right now we have separate files for those so we should try figure out a solution that's consistent.